### ISSUE TITLE
6.3.3 Did not find any CSF globs in preview-stats.json (It was working with 6.2.3)

### ISSUE THREAD

**Issue by VickyKoblinski** [chromaui/chromatic-cli#486]
We're using github actions to add Chromatic to our pipeline. It suddenly stopped working today and after some investigation, the only difference is that the CLI version used by the Github Action bumped up from 6.2.3 to 6.3.3.

```
    → Validating Storybook files
Retrieving story files affected by recent changes
    → Traversing dependencies for 108 files that changed since the last build
✖ Did not find any CSF globs in /tmp/chromatic--1765-AcyfWpX0wn54/preview-stats.json
Check your stories configuration in frontend/.storybook/main.js
Failed to retrieve dependent story files
ℹ Read more at https://storybook.js.org/docs/null/configure/overview
    → Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json

✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json
→ View the full stacktrace below

If you need help, please chat with us at https://www.chromatic.com/docs/cli for the fastest response.
You can also email the team at support@chromatic.com if chat is not an option.

Please provide us with the above CLI output and the following info:

{
  "timestamp": "2021-12-22T21:30:26.090Z",
  "sessionId": "3ba52350-0fdc-486b-aec5-418ba46dbb22",

  "gitVersion": "2.34.1",
  "nodePlatform": "linux",

  "nodeVersion": "12.13.1",
  "packageName": "chromatic",

  "packageVersion": "6.3.3",
  "storybook": {
Error: non-zero exit code
    "viewLayer": null,
    "version": null,
    "addons": []
  },
  "flags": {
    "projectToken": "***",
    "workingDir": "frontend",
    "skip": false,
    "onlyChanged": true,
    "fromCI": true,
    "interactive": false,
    "exitZeroOnChanges": true,
    "exitOnceUploaded": false,
    "allowConsoleErrors": false
  },
  "buildScript": "build-storybook",
  "spawnParams": {
    "client": "npm",
    "clientVersion": "8.1.2",
    "nodeVersion": "v16.13.1",
    "platform": "linux",
    "command": "npm",
    "clientArgs": [
      "run",
      "--silent"
    ],
    "scriptArgs": [
      "build-storybook",
      "--",
      "--output-dir",
      "/tmp/chromatic--1765-AcyfWpX0wn54",
      "--webpack-stats-json",
      "/tmp/chromatic--1765-AcyfWpX0wn54"
    ]
  },
  "errorType": "Error",
  "errorMessage": "✖ Failed to publish your built Storybook",
  "build": {
    "number": 271,
    "webUrl": "https://www.chromatic.com/build?appId=6196dd942f4468003ac25bff&number=271"
  }
}

Error: ✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json
    at /home/runner/work/_actions/chromaui/action/v1/action/main.js:188:51
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:133
    at async e.exports.task (/home/runner/work/_actions/chromaui/action/v1/action/main.js:15:1296)
```

```
# Workflow name
name: "Chromatic"

# Event for the workflow
on:
  # Chromatic needs the commit sha of master to use as image base for comparisons
  push:
    branches:
      - master
  # Only run if the PR contains changes to frontend/**
  pull_request:
    types: [opened, synchronize]

# List of jobs
jobs:
  chromatic-deployment:
    # Operating System
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: frontend
    # Job steps
    steps:
      - name: Check if service has changed
        uses: actions/github-script@0.9.0
        id: service-changed
        with:
          result-encoding: string
          script: |
            if (context.eventName == 'push') return false
            const result = await github.pulls.listFiles({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.number,
              per_page: 100
            })
            const serviceChanged = result.data.filter(f => f.filename.startsWith("frontend/")).length > 0
            console.log(serviceChanged)
            return serviceChanged
      - uses: actions/checkout@v1
      - name: Install dependencies
        run: yarn
        working-directory: frontend
        # Adds Chromatic as a step in the workflow
      - name: Publish to Chromatic
        uses: chromaui/action@v1
        # Chromatic GitHub Action options
        with:
          workingDir: frontend
          token: ${{ secrets.GITHUB_TOKEN }}
          # Chromatic projectToken, refer to the manage page to obtain it.
          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
          onlyChanged: true
          skip: ${{ steps.service-changed.outputs.result == 'false' }}
```

**Comment by VickyKoblinski** [comment]
https://github.com/chromaui/chromatic-cli/issues/467 This may be related.

**Comment by VickyKoblinski** [comment]
Removing `onlyChange: true` fixed this. I guess we don't have TurboSnap configured correctly. :)

**Comment by selenehyun** [comment]
Same problem here!
I added `staticDirs` and posted a PR to use `msw` in chromatic, but suddenly the build started failing.

Until a few hours ago, all PRs failed chromatic builds, but now only PRs with `staticDirs` modified fail chromatic builds.
below is a part of GitHub Action log.
```
Collecting Storybook metadata
Collected Storybook metadata
    → ; no supported addons found
Building your Storybook
    → Running command: npm run --silent build-storybook -- --output-dir /tmp/chromatic--1878-17LOz8oau8bM --webpack-stats-json /tmp/chromatic--1878-17LOz8oau8bM
Storybook built in 1 minute 48 seconds
    → View build log at /home/runner/work/myproject/myproject/build-storybook.log
Publish your built Storybook
    → Validating Storybook files
Retrieving story files affected by recent changes
    → Traversing dependencies for 48 files that changed since the last build
✖ Did not find any CSF globs in /tmp/chromatic--1878-17LOz8oau8bM/preview-stats.json
Check your stories configuration in .storybook/main.js
ℹ Read more at https://storybook.js.org/docs/null/configure/overview
Failed to retrieve dependent story files
    → Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json

✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json
→ View the full stacktrace below

If you need help, please chat with us at https://www.chromatic.com/docs/cli for the fastest response.
You can also email the team at support@chromatic.com if chat is not an option.

Please provide us with the above CLI output and the following info:
{
  "timestamp": "2021-12-23T01:08:48.659Z",
  "sessionId": "b6a4923d-c28f-4fc1-be13-70fe8ec7edd0",
  "gitVersion": "2.34.1",
  "nodePlatform": "linux",
  "nodeVersion": "12.13.1",
  "packageName": "chromatic",
  "packageVersion": "6.3.3",
  "storybook": {
    "viewLayer": null,
    "version": null,
    "addons": []
  },
  "flags": {
    "projectToken": "***",
    "onlyChanged": true,
    "fromCI": true,
    "interactive": false,
    "exitZeroOnChanges": true,
    "exitOnceUploaded": true,
    "allowConsoleErrors": false,
    "ignoreLastBuildOnBranch": "main"
  },
  "buildScript": "build-storybook",
  "spawnParams": {
    "client": "npm",
    "clientVersion": "8.1.2",
    "nodeVersion": "v16.13.1",
    "platform": "linux",
    "command": "npm",
    "clientArgs": [
      "run",
      "--silent"
    ],
    "scriptArgs": [
      "build-storybook",
      "--",
      "--output-dir",
      "/tmp/chromatic--1878-17LOz8oau8bM",
      "--webpack-stats-json",
      "/tmp/chromatic--1878-17LOz8oau8bM"
    ]
  },
  "errorType": "Error",
  "errorMessage": "✖ Failed to publish your built Storybook",
  "build": {
    "number": 1122,
    "webUrl": "https://www.chromatic.com/build?appId=6114bd4e530e1e003b8c1f28&number=1122"
  }
}

Error: ✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json
    at /home/runner/work/_actions/chromaui/action/v1/action/main.js:188:51
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:133
    at async e.exports.task (/home/runner/work/_actions/chromaui/action/v1/action/main.js:15:1296)
```

I build it myself using the chromatic cli on my mac and it deploys just fine. Both 6.2.2 and 6.3.3 work.
But in GitHub Action it seems to work properly only when you disable TurboSnap or don't use `staticDirs`

**Comment by VickyKoblinski** [comment]
Hmm my problem may be a little different. Mine fails when using the cli too, even after removing staticDirs. 

I'm working with a monorepo where my main.js is in `frontend/.storybook/main.js` and I wonder if the code for turbo, when looking at the stories in main.js (`stories: ["../src/**/*.stories.@(js|jsx|mdx)"]`) it's not pathing to the correct relative location? (Like looking into `my_project/src/**` instead of `my_project/frontend/src/**`) The build works fine though. When I pass in the debug flag, I see that it's showing the correct files as modified.

```
  frontend/src/components/RButton/RButton.jsx
  frontend/src/components/RButton/RButton.stories.js
```

But it fails saying `Did not find any CSF globs` when changing either of these should should match the rule `../src/**/*.stories.@(js|jsx|mdx)`.

**Comment by VickyKoblinski** [comment]
Upon further investigation, I found what's missing. 

If I run this, it will work:

`npx chromatic --only-changed --storybook-base-dir frontend`

So the problem I have then is that in the github action, `workingDir` is not equivalent to `--storybook-base-dir` and from the documentation, I don't see an equivalent.

**Comment by anilanar** [comment]
Same issue @userlike, seems 6.3.3 broke TurboSnap.

**Comment by ghengeveld** [comment]
To clarify: we changed TurboSnap to throw when no CSF globs are found because previously it would mean 0 stories got tested and the build would succeed, despite it being misconfigured. Since this would easily go unnoticed, it undermined the reliability of Chromatic as a platform.

If you encounter this error, you should try to set the `--storybook-base-dir` flag (or `storybookBaseDir` with the GitHub Action), or remove `--only-changed` / `onlyChanged`. The base dir is the relative path from the root of your Git repository to the directory where you are executing `build-storybook` or `chromatic` (typically there's a `package.json` with Storybook dependencies there).

With https://github.com/chromaui/chromatic-cli/pull/490 we're going to provide a better error message and suggest the fix.

**Comment by ghengeveld** [comment]
> So the problem I have then is that in the [github action](https://github.com/marketplace/actions/publish-to-chromatic), `workingDir` is not equivalent to `--storybook-base-dir` and from the documentation, I don't see an equivalent.

This is an oversight and needs to be remedied.

**Comment by ghengeveld** [comment]
@VickyKoblinski @anilanar You should now be able to use `storybookBaseDir` with our GitHub Action. I've also added support for `zip` which zips up your Storybook before uploading (which is usually faster).

**Comment by VickyKoblinski** [comment]
> @VickyKoblinski @anilanar You should now be able to use `storybookBaseDir` with our GitHub Action. I've also added support for `zip` which zips up your Storybook before uploading (which is usually faster).

Thank you!! I tried it out now but I'm running into an error. I'm putting in a string but it's spitting out a boolean?

```
Run chromaui/action@v1
  with:
    workingDir: frontend
    token: ***
    projectToken: ***
    onlyChanged: true
    skip: false
    storybookBaseDir: frontend

Chromatic CLI v6.3.3
https://www.chromatic.com/docs/cli

Authenticating with Chromatic
    → Connecting to https://index.chromatic.com
Could not retrieve package info from registry; skipping update check
Authenticated with Chromatic
    → Using project token '********481a'
Retrieving git information
Found 27 changed files:
  .github/workflows/chromatic.yml
  frontend/public/mockServiceWorker.js
  frontend/src/releaseComponents/RButton/RButton.jsx
  frontend/src/stories/Introduction.stories.mdx
  .github/workflows/rollbar_deploy.yml
  CHANGELOG.md
  backend/Gemfile
  backend/Gemfile.lock
  backend/app/controllers/cloud_integrations_controller.rb
  backend/app/interactions/release_installations/register_installation.rb
  backend/app/lib/cloud/aws/ec2.rb
  backend/app/models/cluster.rb
  backend/app/models/clusters_v2/cloud_integration.rb
  backend/app/models/clusters_v2/cloud_integrations/aws.rb
  backend/app/models/clusters_v2/cloud_integrations/google.rb
  backend/config/environments/development.rb
  backend/config/environments/production.rb
  backend/config/initializers/datadog.rb
  backend/db/migrate/20211227192951_add_disabled_to_cloud_integration.rb
  backend/db/structure.sql
  backend/kube/templates/common/autoscaler/aws-k8s-cni.yaml
  backend/kube/templates/common/datadog/datadog-agent-customer.yaml
  backend/kube/templates/common/datadog/datadog-agent.yaml
  backend/kube/templates/common/datadog/datadog-secret-customer.yaml
  backend/kube/templates/common/datadog/datadog-secret.yaml
  backend/kube/templates/common/nginx/mandatory.yaml
  frontend/src/connectedComponents/Instances/ConnectedInstancesTable.js
Retrieved git information
    → Commit 'af0b530' on branch 'storybook-turbo'; found 2 parent builds and 27 changed files
Collecting Storybook metadata
Collected Storybook metadata
    → ; no supported addons found
Building your Storybook
    → Running command: npm run --silent build-storybook -- --output-dir /tmp/chromatic--1815-Pu0Fg7t3raes --webpack-stats-json /tmp/chromatic--1815-Pu0Fg7t3raes
Storybook built in 2 minutes 4 seconds
    → View build log at /home/runner/work/spacedust/spacedust/frontend/build-storybook.log
Publish your built Storybook
    → Validating Storybook files
Retrieving story files affected by recent changes
    → Traversing dependencies for 27 files that changed since the last build
Failed to retrieve dependent story files
    → Could not retrieve dependent story files.
The "to" argument must be of type string. Received type boolean

✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
The "to" argument must be of type string. Received type boolean
→ View the full stacktrace below

If you need help, please chat with us at https://www.chromatic.com/docs/cli for the fastest response.
You can also email the team at support@chromatic.com if chat is not an option.

Please provide us with the above CLI output and the following info:
{
  "timestamp": "2021-12-28T17:18:19.913Z",

  "sessionId": "e545c09c-fb00-46c2-8017-69a12e0c17ca",
  "gitVersion": "2.34.1",
  "nodePlatform": "linux",

  "nodeVersion": "12.13.1",

  "packageName": "chromatic",

Error: non-zero exit code
  "packageVersion": "6.3.3",
  "storybook": {
    "viewLayer": null,
    "version": null,
    "addons": []
  },
  "flags": {
    "projectToken": "***",
    "workingDir": "frontend",
    "skip": false,
    "storybookBaseDir": false,
    "onlyChanged": true,
    "fromCI": true,
    "interactive": false,
    "exitZeroOnChanges": true,
    "exitOnceUploaded": false,
    "allowConsoleErrors": false,
    "zip": false
  },
  "buildScript": "build-storybook",
  "spawnParams": {
    "client": "npm",
    "clientVersion": "8.1.2",
    "nodeVersion": "v16.13.1",
    "platform": "linux",
    "command": "npm",
    "clientArgs": [
      "run",
      "--silent"
    ],
    "scriptArgs": [
      "build-storybook",
      "--",
      "--output-dir",
      "/tmp/chromatic--1815-Pu0Fg7t3raes",
      "--webpack-stats-json",
      "/tmp/chromatic--1815-Pu0Fg7t3raes"
    ]
  },
  "errorType": "TypeError",
  "errorMessage": "✖ Failed to publish your built Storybook",
  "build": {
    "number": 311,
    "webUrl": "https://www.chromatic.com/build?appId=6196dd942f4468003ac25bff&number=311"
  }
}

TypeError [ERR_INVALID_ARG_TYPE]: The "to" argument must be of type string. Received type boolean
    at validateString (internal/validators.js:112:11)
    at Object.relative (path.js:1055:5)
    at /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:379
    at /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:393
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:133
    at async e.exports.task (/home/runner/work/_actions/chromaui/action/v1/action/main.js:15:1296)
```

I suspect `storybookBaseDir: frontend` getting spat out as `"flags": { "storybookBaseDir": false }` is the problem?

**Comment by ghengeveld** [comment]
That seems to be an error originating from [`react-scripts`](https://stackoverflow.com/questions/58199001/typeerror-err-invalid-arg-type-the-path-argument-must-be-of-type-string-re), which is not a dependency of our CLI or the Action, so likely a build error in your Storybook.

**Comment by ghengeveld** [comment]
Could you try to put the value in quotes? So `storybookBaseDir: "frontend"`
Looks like this error originates from `path.relative` which receives `storybookBaseDir`.
If this doesn't work, I recommend you enable diagnostics, [like this](https://github.com/chromaui/chromatic-cli/commit/bf0999da3906fc50f269e6b7573103990c90c765?w=1).

**Comment by VickyKoblinski** [comment]
Alright, I tried both `storybookBaseDir: "frontend"` and `storybookBaseDir: "./frontend"` I have the same error with those.

```
TypeError [ERR_INVALID_ARG_TYPE]: The "to" argument must be of type string. Received type boolean
    at validateString (internal/validators.js:112:11)
    at Object.relative (path.js:1055:5)
    at /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:379
    at /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:393
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:133
    at async e.exports.task (/home/runner/work/_actions/chromaui/action/v1/action/main.js:15:1296)
ℹ Wrote Chromatic diagnostics report to chromatic-diagnostics.json
```

With this yaml

```yaml
# Workflow name
name: "Chromatic"

# Event for the workflow
on:
  # Chromatic needs the commit sha of master to use as image base for comparisons
  push:
    branches:
      - master
  # Only run if the PR contains changes to frontend/**
  pull_request:
    types: [opened, synchronize]

# List of jobs
jobs:
  chromatic-deployment:
    # Operating System
    continue-on-error: true
    runs-on: ubuntu-latest
    defaults:
      run:
        working-directory: frontend
    # Job steps
    steps:
      - name: Check if service has changed
        uses: actions/github-script@0.9.0
        id: service-changed
        with:
          result-encoding: string
          script: |
            if (context.eventName == 'push') return true
            const result = await github.pulls.listFiles({
              owner: context.payload.repository.owner.login,
              repo: context.payload.repository.name,
              pull_number: context.payload.number,
              per_page: 100
            })
            const serviceChanged = result.data.filter(f => f.filename.startsWith("frontend/")).length > 0
            console.log(serviceChanged)
            return serviceChanged

      - uses: actions/checkout@v1
      - name: Install dependencies
        run: yarn
        working-directory: frontend
        # Adds Chromatic as a step in the workflow
      - name: Publish to Chromatic
        uses: chromaui/action@v1
        continue-on-error: true
        # Chromatic GitHub Action options
        with:
          workingDir: frontend
          token: ${{ secrets.GITHUB_TOKEN }}
          # Chromatic projectToken, refer to the manage page to obtain it.
          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
          onlyChanged: true
          skip: ${{ steps.service-changed.outputs.result == 'false' }}
          storybookBaseDir: "frontend"
          zip: true
          exitZeroOnChanges: true
          exitOnceUploaded: true
          diagnostics: true
      - name: print test results
        run: cat chromatic-diagnostics.json
      - uses: actions/upload-artifact@v2
        with:
          name: chromatic-diagnostics
          path: chromatic-diagnostics.json
```

I don't see anything useful in the diagnostic output, but here it is:

https://gist.github.com/VickyKoblinski/81abe8044a17880865ec60a20b522134

**Comment by larrifax** [comment]
I'm getting the same error as you, albeit with a different config (webpack5, storyStoreV7, CSFv3 and `stories: ['../packages/ui/src']`).

In my case, `yarn chromatic trim-stats-file apps/storybook/storybook-static/preview-stats.json` produces a file where there's only a single module entry matching the condition in https://github.com/chromaui/chromatic-cli/blob/main/bin-src/lib/getDependentStoryFiles.js#L93, and that's the following:
```
    {
      "id": "../../packages/ui/src lazy recursive ^\\.\\/.*$ include: (?:\\/packages\\/ui\\/src(?:\\/(?%21\\.)(?:(?:(?%21(?:^|\\/)\\.).)*?)\\/|\\/|$)(?%21\\.)(?=.)[^/]*?\\.stories\\.(mdx|tsx|ts|jsx|js))$",
      "name": "../../packages/ui/src/ lazy ^\\.\\/.*$ include: (?:\\/packages\\/ui\\/src(?:\\/(?%21\\.)(?:(?:(?%21(?:^|\\/)\\.).)*?)\\/|\\/|$)(?%21\\.)(?=.)[^/]*?\\.stories\\.(mdx|tsx|ts|jsx|js))$ chunkName: [request] namespace object",
      "reasons": [
        { "moduleName": "./storybook-config-entry.js + 1 modules" },
        { "moduleName": "./storybook-stories.js" }
      ]
    },
```
I don't see any logic supporting this kind of `module.name`, so I guess my case just isn't supported yet.

**Comment by kylegach** [comment]
Hi, folks. If I did it right (my first time publishing the action 😅 ), the `storybookBaseDir` option should work correctly now. Let us know if you still have any problems!

**Comment by kylegach** [comment]
@larrifax — Although the title of this issue would seem to fit your problem, I think that's a separate issue and would be best handled independent of this conversation. Mind creating a new issue?

**Comment by VickyKoblinski** [comment]
Yesss, the `storybookBaseDir` is showing up correctly now. But I'm running into another error, `Did not find any CSF globs in preview-stats.json`

```
Chromatic CLI v6.3.3
https://www.chromatic.com/docs/cli

Authenticating with Chromatic
    → Connecting to https://index.chromatic.com
Could not retrieve package info from registry; skipping update check
Authenticated with Chromatic
    → Using project token '********481a'
Retrieving git information
Found 3 changed files:
  .github/workflows/chromatic.yml
  frontend/public/mockServiceWorker.js
  frontend/src/releaseComponents/RButton/RButton.jsx
Retrieved git information
    → Commit '8bf828d' on branch 'storybook-turbo-2'; found 1 parent build and 3 changed files
Collecting Storybook metadata
Collected Storybook metadata
    → ; no supported addons found
Building your Storybook
    → Running command: npm run --silent build-storybook -- --output-dir /tmp/chromatic--1874-ztV3NmkFRShn --webpack-stats-json /tmp/chromatic--1874-ztV3NmkFRShn
Storybook built in 2 minutes 51 seconds
    → View build log at /home/runner/work/spacedust/spacedust/frontend/build-storybook.log
Publish your built Storybook
    → Validating Storybook files
Retrieving story files affected by recent changes
    → Traversing dependencies for 3 files that changed since the last build
✖ Did not find any CSF globs in /tmp/chromatic--1874-ztV3NmkFRShn/preview-stats.json
Check your stories configuration in frontend/frontend/.storybook/main.js
ℹ Read more at https://storybook.js.org/docs/null/configure/overview
Failed to retrieve dependent story files
    → Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json

✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json
→ View the full stacktrace below

If you need help, please chat with us at https://www.chromatic.com/docs/cli for the fastest response.
You can also email the team at support@chromatic.com if chat is not an option.

Please provide us with the above CLI output and the following info:
{
  "timestamp": "2022-01-03T21:57:38.297Z",
  "sessionId": "f5545e80-194e-4735-9870-c5c337d5321f",
  "gitVersion": "2.34.1",
  "nodePlatform": "linux",
  "nodeVersion": "12.13.1",
  "packageName": "chromatic",
  "packageVersion": "6.3.3",
  "storybook": {
    "viewLayer": null,
    "version": null,
    "addons": []
  },
  "flags": {
    "projectToken": "***",
    "workingDir": "frontend",
    "diagnostics": true,
    "skip": false,
    "storybookBaseDir": "frontend",
    "onlyChanged": true,
    "fromCI": true,
    "interactive": false,
    "exitZeroOnChanges": true,
    "exitOnceUploaded": true,
    "allowConsoleErrors": false,
    "zip": true
  },
  "buildScript": "build-storybook --quiet",
  "spawnParams": {
    "client": "npm",
    "clientVersion": "8.1.2",
    "nodeVersion": "v16.13.1",
    "platform": "linux",
    "command": "npm",
    "clientArgs": [
      "run",
      "--silent"
    ],
    "scriptArgs": [
      "build-storybook",
      "--",
      "--output-dir",
      "/tmp/chromatic--1874-ztV3NmkFRShn",
      "--webpack-stats-json",
      "/tmp/chromatic--1874-ztV3NmkFRShn"
    ]
  },
  "errorType": "Error",
  "errorMessage": "✖ Failed to publish your built Storybook",
  "build": {
    "number": 380,
    "webUrl": "https://www.chromatic.com/build?appId=6196dd942f4468003ac25bff&number=380"
  }
}

Error: ✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json
    at /home/runner/work/_actions/chromaui/action/v1/action/main.js:188:51
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/chromaui/action/v1/action/main.js:184:133
    at async e.exports.task (/home/runner/work/_actions/chromaui/action/v1/action/main.js:15:1296)
ℹ Wrote Chromatic diagnostics report to chromatic-diagnostics.json
```

I ran into this same issue before when I added `storyStoreV7` to my `main.js` so it might be related? (To be clear, I don't have `storyStoreV7` enabled right now, I had to remove it.) The yaml is still the same.

**Comment by tmeasday** [comment]
Hey @VickyKoblinski, just to jump in for @ghengeveld here as he is away this week. We might need to take a look at your webpack stats file here to debug this. I'll write the steps here in case it helps others.

1. Firstly, you probably want to trim the stats file first, it is huge to begin with and hard to work with: https://www.chromatic.com/docs/turbosnap.html#turbosnap
2. Next, pick a story file, say `Button.stories.ts`, that should be picked up by a `stories` glob
3. Find it in the stats file. You are looking for a "module" that includes that file [1]
3.a if you can't find the story file, check a dependency of it (e.g the actual component it references). Is it in there? What are the "reasons" for it to be there?
4.  Check the "reasons" listed for that story file? (i.e. why it was included in the webspack build)? It should be a module that looks like a mess of regular expression code, like `"./src/components sync ^\\.(?:(?:^|\\/|(?:(?:(?!(?:^|\\/)\\.).)*?)\\/)(?!\\.)(?=.)[^/]*?\\.stories\\.js)$"`
5. Find *that* module in the stats file. [2] What is its reasons? Does it match the list here: https://github.com/chromaui/chromatic-cli/blob/bdda9b4370f00454ba6c475e7ee84385a60e123b/bin-src/lib/getDependentStoryFiles.js#L64-L71?


[1] an example:

```
{
      "id": "./src/components/A.stories.js",
      "name": "./src/components/A.stories.js + 1 modules",
      "modules": [
        { "name": "./src/components/A.stories.js" },
        { "name": "./src/components/A.js" }
      ],
      "reasons": [
        { "moduleName": "./src/components sync ^\\.(?:(?:^|\\/|(?:(?:(?!(?:^|\\/)\\.).)*?)\\/)(?!\\.)(?=.)[^/]*?\\.stories\\.js)$" }
      ]
    },
```

[2] same example:

```
    {
      "id": "./src/components sync recursive ^\\.(?:(?:^|\\/|(?:(?:(?!(?:^|\\/)\\.).)*?)\\/)(?!\\.)(?=.)[^/]*?\\.stories\\.js)$",
      "name": "./src/components sync ^\\.(?:(?:^|\\/|(?:(?:(?!(?:^|\\/)\\.).)*?)\\/)(?!\\.)(?=.)[^/]*?\\.stories\\.js)$",
      "reasons": [
        { "moduleName": "./generated-stories-entry.js" }
      ]
    },
```

**Comment by danguilherme** [comment]
I am using Storybook v6.5.15 and I also faced this issue. Disabling `storyStoreV7` also worked for me.

**Comment by markwoon** [comment]
Removing `storyStoreV7` unfortunately did not fix problem for me wiht 6.5.16:

```
{
  "timestamp": "2023-09-10T20:49:12.604Z",
  "sessionId": "0f3f1e8c-c9f7-417b-b30f-f2afc58a464f",
  "gitVersion": "2.34.1",
  "nodePlatform": "linux",
  "nodeVersion": "18.16.0",
  "packageName": "chromatic",
  "packageVersion": "6.19.9",
  "storybook": {
    "viewLayer": "react",
    "version": "6.5.16",
    "builder": {
      "name": "webpack5",
      "packageVersion": "6.5.16"
    },
    "addons": [
      {
        "name": "a11y",
        "packageName": "@storybook/addon-a11y",
        "packageVersion": "6.5.16"
      },
      {
        "name": "essentials",
        "packageName": "@storybook/addon-essentials",
        "packageVersion": "6.5.16"
      },
      {
        "name": "interactions",
        "packageName": "@storybook/addon-interactions",
        "packageVersion": "6.5.16"
      },
      {
        "name": "storysource",
        "packageName": "@storybook/addon-storysource",
        "packageVersion": "6.5.16"
      }
    ]
  },
  "flags": {
    "c": true,
    "i": true,
    "exitZeroOnChanges": "",
    "onlyChanged": "development",
    "storybookBuildDir": [
      "build/storybook"
    ],
    "projectToken": [],
    "outputDir": [],
    "externals": [],
    "onlyStoryFiles": [],
    "onlyStoryNames": [],
    "untraced": [],
    "interactive": true,
    "appCode": []
  },
  "exitCode": 0,
  "exitCodeKey": "OK",
  "errorType": "Error",
  "errorMessage": "✖ Failed to publish your built Storybook",
  "build": {
    "id": "64efd8bc7e034f98a9b2f648",
    "number": 697
  }
}

Error: ✖ Failed to publish your built Storybook
Could not retrieve dependent story files.
Did not find any CSF globs in preview-stats.json
```

What exactly is this CSF glob that it is looking for?

**Comment by paulelliott** [comment]
Closed due to age. Please reopen if you feel this is still relevant for you.
